components/docs-chef-io/content/habitat/origin-keys.md

```diff
@@ -0,0 +1,190 @@
+
+Prerequisites:
+
+- [Get Chef Habitat]({{< relref "install-habitat.md" >}})
+- [Create a Chef Habitat Builder account]({{< relref "builder-account#builder-account" >}})
+- [Generate a personal access token]({{< relref "builder-profile#create-a-personal-access-token" >}})
+- [Create an origin]({{< relref "origins" >}}) or join an origin by [invitation]({{< relref "origin-rbac#manage-origin-membership-with-hab-origin-invitations" >}})
+
+When you create an origin, Chef Habitat Builder automatically generates _origin keys_.
+Origin key cryptography is asymmetric: it has a public origin key that you can distribute freely, and a private origin key that you should distribute only to users of the origin.
+All Chef Habitat Builder users with access to the origin can view the origin public key revisions in the origin key tab (Builder > Origin > Keys) and download the origin public key, but only users with the origin 'administrator' or 'owner' roles can view or download the origin private key, or change the origin key pair.
+
+| Keys Actions | Read-Only | Member | Maintainer | Administrator | Owner |
+|---------|-------|-------|-------|-------|-------|
+| View keys | Y | Y | Y | Y | Y |
+| Add/Update/Delete keys | N | N | N | Y | Y |
+
+Chef Habitat uses origin keys:
+
+- When you build an artifact in your local environment, Chef Habitat signs the artifact with a public key
+- When you upload an artifact to Chef Habitat Builder or Builder on-prem, Chef Habitat verifies that the artifact was signed with its public key
+- When you install an artifact on a Chef Habitat Supervisor, Chef Habitat uses the public origin key to authorize the artifact's installation; Chef Habitat only installs artifacts for which it has the public origin key
+- When you download an artifact to your local environment, Chef Habitat uses the public origin key to verify the artifact's integrity before it starts the installation
+
+Chef Habitat Builder origin key names follow the format:
+
+```hab
+<origin>-<datetime>.pub (public key)
+<origin>-<datetime>.sig.key (private key, also called a "signing key")
+```
+
+For example, in:
+
+```hab
+testorigin-20190416223046.pub
+testorigin-20190416223046.sig.key
+```
+
+- "testorigin" is the origin's name
+- "20190416223046" is the date and time of the key's creation, which was 2019-04-16 22:30:46.
+- `.pub` is the file extension for the public key
+- `.sig.key` is the file extension for the private key, which is also called a "signing key"
+
+## The Keys Tab
+
+When you create an origin, Chef Habitat Builder automatically generates an origin key pair and saves both keys. To view your origin keys on Chef Habitat Builder, navigate to your origin and select the **Keys** tab. (Builder > Origins > Keys) You will always be able to view and download origin public keys, but you will only see the private keys for origins in which you are an "administrator" or "owner".
```

Comment:
It has not always been very clear, but there are actually _two_ distinct "origin key pairs" floating around. The first one, and the one described in this section, is the signing key pair. The second one came along much later, and is an encryption key pair, used for safely uploading build secrets to Builder. Our CLI and documentation hasn't ever really been direct about which is which, unfortunately. As such, it would be nice to start being more specific in this documentation.

Reply:
I have a plan for this!!! But not in this PR, because we need a separate PR to adequately represent your key work.